plugin/catalog-plugin-jpeg2000-thumbnail-converter/src/main/java/ddf/catalog/plugin/jpeg2000/thumbnail/converter/Jpeg2000ThumbnailConverter.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+package ddf.catalog.plugin.jpeg2000.thumbnail.converter;
+
+import com.sun.media.imageioimpl.plugins.jpeg2000.IISRandomAccessIO;
+import com.sun.media.imageioimpl.plugins.jpeg2000.J2KImageReaderSpi;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+import javax.imageio.ImageIO;
+import javax.imageio.spi.IIORegistry;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+/**
+ * check for Jpeg 2000 thumbnails in the result set, and convert them
+ * to standard Jpeg so the browser can render them.
+ */
+public class Jpeg2000ThumbnailConverter implements PostQueryPlugin {
+
+    public static final int OTHER_JP2_SIGNATURE = 0x0000000c;
+
+    public static final int JP2_SIGNATURE_BOX = 0x6a502020;
+
+    public static final int OFFICIAL_JP2_SIGNATURE = 0x0d0a870a;
+
+    public static final short START_OF_CODESTREAM_MARKER = (short) 0xff4f;
+
+    public Jpeg2000ThumbnailConverter() {
+        IIORegistry.getDefaultInstance().registerServiceProvider(new J2KImageReaderSpi());
+    }
+
+    @Override
+    public QueryResponse process(QueryResponse input)
+            throws PluginExecutionException, StopProcessingException {
+        for (Result result : input.getResults()) {
+            Metacard metacard = result.getMetacard();
+            byte[] thumbnailBytes = metacard.getThumbnail();
+            if (thumbnailBytes == null)
+                continue;
+
+            try (
+                    ByteArrayInputStream original = new ByteArrayInputStream(thumbnailBytes);
+                    ByteArrayOutputStream converted = new ByteArrayOutputStream()
+            ) {
+                IISRandomAccessIO in = new IISRandomAccessIO(
+                        ImageIO.createImageInputStream(original));
+                // extracted from jj2000.j2k.fileformat.reader.FileFormatReader
```

Comment:
OK thanks. I was thinking more about actually listing out the copyright for the code as it was copied over from another copyrighted source. For compatible licenses I believe I have seen that done in the main header of the source file. That's really what I was questioning here, that there was no copyright attribution to the original author :)